src/main/java/com/byteowls/vaadin/chartjs/ChartJs.java

```diff
@@ -2,18 +2,28 @@
 
 import com.byteowls.vaadin.chartjs.config.ChartConfig;
 import com.vaadin.annotations.JavaScript;
+import com.vaadin.annotations.StyleSheet;
+import com.vaadin.icons.VaadinIcons;
 import com.vaadin.ui.AbstractJavaScriptComponent;
 import com.vaadin.ui.JavaScriptFunction;
 import elemental.json.JsonArray;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
 
 @JavaScript({"vaadin://chartjs/Moment.js", "vaadin://chartjs/Chart.min.js", "vaadin://chartjs/hammer.min.js", "vaadin://chartjs/chartjs-plugin-zoom.min.js",
     "vaadin://chartjs/chartjs-plugin-annotation.min.js", "vaadin://chartjs/chartjs-connector.js"})
+@StyleSheet("vaadin://chartjs/chartjs-connector.css")
 public class ChartJs extends AbstractJavaScriptComponent {
 
     private static final long serialVersionUID = 2999562112373836140L;
+    /**
+     * Counter for the next menu action being added, used to create a unique
+     * property name.
+     */
+    private static final AtomicInteger nextMenuId = new AtomicInteger(0);
```

Comment:
It makes that part thread-safe. So if the code is used concurrently from multiple threads, using incrementAndGet will always return an integer value which has never been used by a different thread. I think the field has to be static, since the integer is also used as identifier on the browser side, so multiple chart instances need to use different values. Otherwise there could be problems if two chart instances are displayed at once.

Reply:
Ok. That's probably sth the rest of the addon needs as well.